### PR TITLE
Improved reu size detection.

### PR DESCRIPTION
**Pull request by blackystardust**
Improved reu size detection.

### CODE REVIEW

File: libsrc/c128/emd/c128-reu.s [hunk 1]
```diff
@@ -92,17 +92,56 @@ INSTALL:
         cmp     REU_REUADDR             ; Check for presence of REU
         bne     nodevice
 
-        ldy     #>(128*4)               ; Assume 128KB
-        lda     REU_STATUS
-        and     #$10                    ; Check size bit
-        beq     @L1
-        ldy     #>(256*4)               ; 256KB when size bit is set
-@L1:    sty     pagecount+1
-
```
Comment (mrdudz) [inline libsrc/c128/emd/c128-reu.s]: this is good - the "size" bit isnt actually related to size at all, its the status of the jumper that tells the REC chip what kind of RAM is installed

File: libsrc/c128/emd/c128-reu.s [hunk 2]
```diff
@@ -92,17 +92,56 @@ INSTALL:
...
+        php
+        sei
         ldy     #$FF
-        sty     curpage
-        sty     curpage+1               ; Invalidate the current page
-        txa                             ; X = A = EM_ERR_OK
+loop:
+        sty     window
+        jsr     reu_size_check_common
+        ldx     #%10110000
```
Comment (mrdudz) [inline libsrc/c128/emd/c128-reu.s]: this should use a contant - like in the rest of the driver

File: libsrc/c128/emd/c128-reu.s [hunk 3]
```diff
@@ -92,17 +92,56 @@ INSTALL:
...
+        stx     REU_COMMAND
+        dey
+        cpy     #$FF
+        bne     loop
+        iny
+size_loop:
+        jsr     reu_size_check_common
+        ldx     #%10110001
```
Comment (mrdudz) [inline libsrc/c128/emd/c128-reu.s]: dito - should be a constant

**Comment by blackystardust** [comment]
Replaced reu commands with defined constants.

**Comment by oliverschmidt** [comment]
I don't know if it actually fits here but would like to just in case point to https://github.com/cc65/wiki/wiki/Reminders#C128_library saying: _Test EM drivers with the multidemo sample; and, fix the broken ones._

**Comment by blackystardust** [comment]
@oliverschmidt 

I'll see what I can do, in the vice we have an emd test program with a menu and .d64 image so seperate emd's can be tested on multiple systems. I'll run through the drivers, fix the broken ones and add missing ones.

**Comment by oliverschmidt** [comment]
That sounds great. However, I didn't want to express an expectation with my recent comment. I just wanted to point out that the C128 EMD drivers are supposed/known to have bugs. At least I didn't manage to get them working as expected with VICE x128. In contrast the C64 EMD drivers seemed to work fine with VICE x64.

**Comment by blackystardust** [comment]
@oliverschmidt 

Sofar I've tested the c128 reu emd driver, as fixed in this pr, and it works correctly, I don't know about the others, but will find out.

**Comment by mrdudz** [comment]
so, merge? :)

**Comment by oliverschmidt** [comment]
> so, merge? :)

I'm sure you know all the criteria I consider relevant. So go ahead as discussed if you see fit.

**Comment by blackystardust** [comment]
Updated documentation for the reu emd.

**Comment by mrdudz** [comment]
ok, i dont see any further problems

**Pull request merged**